### ISSUE TITLE
Trakt sync dialogue box never closes

### ISSUE THREAD

**Issue by JustAskDave** [ApolloTVofficial/kamino#92]
When first sync with Trakt, the dialogue box says it will close when finished. Have waited several minutes on both Samsung Galaxy TabA and S9+. Both sit there never closing. Hitting back button manually closes then check and all favourites have been synched sucessfully.

**Comment by edmundackah** [comment]
How many items do you have on trakt?

**Comment by FlyPigsFly** [comment]
I have this same issue. I have over 27k episodes and over 700 movies collected

**Comment by JustAskDave** [comment]
Good question.  Just went to have a look and I have 2,223 episodes (62 shows) and
47 movies collected.  So not really that much I would have thought.  I will try again and leave it longer this time. Though with the predecessor program it used to take less than 30 seconds.

**Comment by JustAskDave** [comment]
Ok, today it is closing after approx 45 sec on both devices. Must have been server issues?  All good.

**Comment by FlyPigsFly** [comment]
We'll wait a minute this issue still exists for me. Syncing for a day now...

**Comment by JustAskDave** [comment]
Reopened to continue monitoring

**Comment by edmundackah** [comment]
What device are you on @FlyPigsFly

**Comment by FlyPigsFly** [comment]
Googles Pixel 2

**Comment by kevincosmos** [comment]
@FlyPigsFly can you verify if this is still occuring?

**Comment by FlyPigsFly** [comment]
Yes still exists. In fact after sync with trakt for about thirty minutes I forced closed the app to rid the sync dialog and nothing appears in favorites. Prior to version 1.3.3 if I forced closed app to rid sync dialog there would be some favorites synced.

**Comment by kevincosmos** [comment]
@FlyPigsFly you do appear to have a lot of episodes, can you confirm how long other apps take to sync these?

If you have the technical knowledge, could you ADB connec your computer to your phone and run adb catlog (If you have a Linux or macOS machine, you should do adb logcat | grep flutter), this will allow the devs to look at the logs and hopefully identify any issue

**Comment by FlyPigsFly** [comment]
TerrariumTV/CyberFlix taking less then 60 seconds, tvzion takes a few minutes and teatv takes less then sixty seconds but their implementation seems to restrict the amount synced for each sync cycle meaning I would have to manually sync n number of times before full sync occurs. I can do as you requested and send the output. Can I send output directly to someone on Apollo team?

**Comment by kevincosmos** [comment]
@FlyPigsFlYeah that would be great, you can message one of the Developers on Discord if you're a member there or send it to us via Twitter DM or on our Facebook Group.

**Comment by FlyPigsFly** [comment]
@kevincosmos I'm unable to provide the requested information... `adb logcat` produces information and I can `adb logcat |grep kamino` successfully but not `adb logcat |grep flutter` Any suggestions maybe I missed something...

**Comment by kevincosmos** [comment]
@FlyPigsFly just provide what you can and hopefully it will be enough :)

**Comment by edmundackah** [comment]
I have looked at the issue, we can impose an initial limit like other apps and force the user to sync later. I can't make the database write any faster.

**Comment by FlyPigsFly** [comment]
Impose a limit would be better then infinite dialog. Could we at least have an option when enabled tells Apollo to auto sync with trakt on start up?

**Comment by kevincosmos** [comment]
I think TVZion's approach works well - by default, Trakt will only copy the previous 8 weeks of history to the client. This can be altered in the settings if you need to sync more. They also allow you to choose what you want to sync on startup (favourites, watch history, etc).

**Comment by polishboi** [comment]
I don't know how SeriesGuide does it but there it's just a click and done. Dunno if it's done in the background or not syncing anything of size though.

**Comment by NBTX** [comment]
Okay, so it seems that we need to restrict the amount of favorites that are pulled in from Trakt - or maybe start the process and carry on in the background. Either way, I'll update the issue title to reflect this.
